float32 all_force
bool stance_flg